### PR TITLE
RUMM-755 Update build, test and code analyse tools

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -38,7 +38,7 @@ dependencies {
     implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.61")
     implementation("io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.1.1")
     implementation("org.jlleitschuh.gradle:ktlint-gradle:9.1.0")
-    implementation("com.android.tools.build:gradle:4.0.0")
+    implementation("com.android.tools.build:gradle:4.0.1")
     implementation("com.github.ben-manes:gradle-versions-plugin:0.27.0")
     implementation("me.xdrop:fuzzywuzzy:1.2.0")
     implementation("org.jetbrains.dokka:dokka-gradle-plugin:0.10.0")
```
Comment (xgouchet) [inline buildSrc/build.gradle.kts:44]: You need to match this with Dokka's version below: `1.4.10`

File: buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -38,7 +38,7 @@ dependencies {
     implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.61")
```
Comment (xgouchet) [inline buildSrc/build.gradle.kts:38]: You need to match this with Kotlin's version below: `1.4.0`

File: buildSrc/build.gradle.kts [hunk 3]
```diff
@@ -38,7 +38,7 @@ dependencies {
...
     implementation("io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.1.1")
     implementation("org.jlleitschuh.gradle:ktlint-gradle:9.1.0")
```
Comment (xgouchet) [inline buildSrc/build.gradle.kts:40]: You need to match this with KtLint's version below: `9.4.0`

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: buildSrc/build.gradle.kts (hunk 2 above)
Comment (mariusc83) [reply buildSrc/build.gradle.kts:38]: I cannot do this as the `kotlin-dsl` plugin still requires `1.3.61`

**Review by xgouchet — APPROVED** [review]

**Pull request merged**